go/enclave/rpc_server.go

```diff
@@ -125,6 +126,16 @@ func (s *server) SubmitBlock(_ context.Context, request *generated.SubmitBlockRe
 	bl := s.decodeBlock(request.EncodedBlock)
 	blockSubmissionResponse, err := s.enclave.SubmitBlock(bl, request.IsLatest)
 	if err != nil {
+		var rejErr common.BlockRejectError
+		isReject := errors.As(err, &rejErr)
+		if isReject {
+			// todo: we should avoid errors in response messages and use the gRPC error objects for this stuff (standardized across all enclave responses)
+			msg, err := rpc.ToBlockSubmissionRejectionMsg(&rejErr)
```

Comment:
Is this logic correct? If we fail to convert to a block submission rejection message, we ignore the resulting error, and fall back to reporting the old error?

At a minimum, we should log the failed conversion to a block submission rejection message, I think.

Reply:
Yeah fair point, I'll log it clearly. It's very weird scenario because we already confirmed that it was a block rejection error.